In [1]:
import pandas as pd

### Data Scraping:
Below are the notebooks we used for scraping data, all of them are done using script instead of manually downloading. \
\
Web Scraping for Properties \
Web Scraping for Shopping Centres \
Web Scraping for Suburbs-LGA-SA2 Mapping \
ABS Dataset \
Find Hospital and Entertainment Facilities Dataset 

### Preprocessing:
Below are the notebooks for preprocessing, mainly deal with missing values, and use techniques to extract useful information from raw data \
\
Preprocessing of Income \
Preprocessing of Population \
Preprocessing of Property Dataset \
Preprocessing of School Locations \
crime_rate \
Preprocess_crime_rate \
preprocess_traffic \
Preprocess for annual rent 

### Feature Engineering:
Calculate Straight Distances \
Count for Facilites, Hospital, School by suburb \
Crime each suburb \
Crime traffic, and income in suburbs \
Merge dataset for Q2 \
Merge datasets for liveable suburbs evaluation \
Shopping central in suburbs \
Use API to Get Route Distance(irreproducible)

### Modelling:
#### Q1:
- Plot the distributions and correlation of attributes
- Apply data transformations
- Use stepwise selection to build statistic model

In [6]:
# Result
q1result = pd.read_csv('../data/curated/q1_result.csv',  index_col=0)
q1result.round(4)

Estimate
(Intercept)              216.8039
Apartment / Unit / Flat   34.3659
House                     72.2002
Studio                   -74.0134
Townhouse                115.4014
Villa                     68.8498
Beds                      46.0708
Baths                     19.5134
numStation_1km             4.9595
numShopping_3km            4.2891
numFacility_3km            0.7006
cbd_route                 -0.0131

- Base price for a property is 216.8039 dollars
- Price increase or decrease based on its property type
- Both number of Beds and Bathrooms increase rental price
- Station, shopping centre, entertainment facility in surrounding area increase price
- the farer from cbd, the lower price is

#### Q2:
- Use rental price, population, income to measure suburb growth
- Build time series model using Auto Regression, use data from earliest recorded time to 2018 for consistency
- Predict values of these data for each suburb in 2025 and rank them
- Use the average rank of three attributes as the final ranking of suburb

In [7]:
# Result
q2result = pd.read_csv('../data/curated/suburb_ranking.csv')

In [9]:
# Top 10 suburbs
q2result[:10]

Suburb  rental rank  income rank  population rank  average rank
0         DROMANA           11           29               10     16.666667
1         PORTSEA           12           54               16     27.333333
2       ASPENDALE           24           16               50     30.000000
3       GROVEDALE           54           34                3     30.333333
4        WERRIBEE           83           10                2     31.666667
5  PORT MELBOURNE           96            1                1     32.666667
6         CHELSEA           23           13               62     32.666667
7       SPOTSWOOD           19           31               54     34.666667
8         NEWPORT           20           30               55     35.000000
9         TORQUAY            6           91               12     36.333333